src/results.cpp

```diff
@@ -259,8 +259,12 @@ void Results::evaluate_query_if_needed(bool wants_notifications)
             return;
         case Mode::Query:
             m_query.sync_view_if_needed();
-            m_table_view = m_query.find_all();
-            if (!m_descriptor_ordering.is_empty()) {
+            if (m_descriptor_ordering.is_empty()) {
+                auto limit = m_descriptor_ordering.has_limit() ? m_descriptor_ordering.get_limit() : size_t(-1);
+                m_table_view = m_query.find_all(0, -1, limit);
+            }
+            else {
+                m_table_view = m_query.find_all();
                 m_table_view.apply_descriptor_ordering(m_descriptor_ordering);
```

Comment:
It looks like having versions of `Query::count` and `Query::find_all` which take a `DescriptorOrdering` would be useful. Having the latter case would also remove the inefficiency of having to run the query twice here, once for `find_all` and having to sync it again in `apply_descriptor_ordering`.